web/main.dart

```diff
@@ -0,0 +1,82 @@
+import 'dart:html';
+import 'dart:svg' as svg;
+import 'dart:convert' show JSON;
+
+import 'package:svg_pan_zoom/svg_pan_zoom.dart' as panzoom;
+
+void main() {
+  querySelector('#path-button').onMouseUp.listen(buttonPressHandler);
+
+  var panZoom = new panzoom.SvgPanZoom.selector('.inner');
+  panZoom
+    ..zoomEnabled = true
+    ..panEnabled = true
+    ..zoomSensitivity = 0.02;
+}
+
+void buttonPressHandler(MouseEvent event) {
+  InputElement input = querySelector('#path-to-json');
+  String pathToJson = input.value;
+  HttpRequest.getString(pathToJson).then((String response) {
+    List data = JSON.decode(response);
+    displayData(data);
+  });
+}
+
+void displayData(List data) {
+  svg.SvgSvgElement innerSvg = querySelector('.inner');
+  svg.GElement group = innerSvg.querySelector('#viewport');
+  group.nodes.clear();
+
+  // Record maximum values of x and y which represent the width and height for
+  // background necessary for zooming and panning.
+  num maxX, maxY;
+  maxX = maxY = double.NEGATIVE_INFINITY;
+
+  for (Map dataPoint in data) {
```

Comment:
Could you add a sanity check on the format of dataPoint here, so that we get debug data that makes sense fi there's a failure?